### ISSUE TITLE
Understanding Code issue! :-) ---- Loss fn

### ISSUE THREAD

**Issue by Sayantan17** [MIC-DKFZ/nnUNet#356]
Hi Fabian @FabianIsensee 

Can you please help me in understanding why you did them? (One by one). Will be great if you help.

In get_tp_fp_fn function, why you did this?
Started the tuple from 2 and the next lines, I mean you checking masks are returning proper identity without squares, then can you please explain x_i * with all rows + columns of the mask?
`
axes = tuple(range(2, len(net_output.size()))) ...... tp = torch.stack(tuple(x_i * mask[:, 0] for x_i in torch.unbind(tp, dim=1)), dim=1)
`

In soft dice loss, can you please explain?
`
if self.batch_dice: axes = [0] + list(range(2, len(shp_x))) else: axes = list(range(2, len(shp_x))) if self.apply_nonlin is not None: x = self.apply_nonlin(x)
`

**Comment by Sayantan17** [comment]
Fabian?
